tests/main/component/task.yaml

```diff
@@ -15,23 +15,38 @@ execute: |
       exit 1
   fi
 
-  # Install snap and component
+  # Install snap
   snap install --dangerous snap-with-comps_1.0_all.snap
-  chg_id=$(snap install --no-wait --dangerous snap-with-comps+comp1_1.0.comp)
-  snap watch "$chg_id"
 
-  # Chech component install change was as expected
-  snap change "$chg_id" | MATCH "^Done .*Prepare component"
-  snap change "$chg_id" | MATCH "^Done .*Mount component"
-  snap change "$chg_id" | MATCH "^Done .*Make component .* available to the system"
-
-  # File has been copied around
-  comp_inst_path=/var/lib/snapd/snaps/snap-with-comps+comp1_x1.comp
-  stat "$comp_inst_path"
-
-  # Component is mounted (note that we need to escape the "+" in the path)
-  mount | MATCH "^${comp_inst_path/+/\\+} on /snap/snap-with-comps/components/x1/comp1.*"
-  # and is seen from snap app
-  snap-with-comps.test
-
-  # TODO: refresh and remove checks when implemented by snapd
+  # Install local component function
+  # $1: expected component revision
+  install_comp() {
+      comp_rev=$1
+      chg_id=$(snap install --no-wait --dangerous snap-with-comps+comp1_1.0.comp)
+      snap watch "$chg_id"
+
+      # Check component install change was as expected
+      snap change "$chg_id" | MATCH "^Done .*Prepare component"
+      snap change "$chg_id" | MATCH "^Done .*Mount component"
+      snap change "$chg_id" | MATCH "^Done .*Make component .* available to the system"
+
+      # File has been copied around
+      comp_inst_path=/var/lib/snapd/snaps/snap-with-comps+comp1_${comp_rev}.comp
+      stat "$comp_inst_path"
+
+      # Component is mounted (note that we need to escape the "+" in the path)
+      # x1 is the snap revision
+      symlink=/snap/snap-with-comps/components/x1/comp1
+      mnt_point=${symlink}_${comp_rev}
+      mount | MATCH "^${comp_inst_path/+/\\+} on ${mnt_point} .*"
+      # And symlinked
+      readlink "$symlink" | MATCH "comp1_$comp_rev"
+      # and is seen from snap app
+      snap-with-comps.test
+  }
+
+  # Install, then reinstall local component
+  install_comp x1
+  install_comp x2
+
+  # TODO: remove checks when implemented by snapd
```

Comment:
Which checks?

Reply:
I've rephrased this, was not really talking about removing test checks :slightly_smiling_face: